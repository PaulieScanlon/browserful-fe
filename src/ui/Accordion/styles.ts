import styled from 'react-emotion';
import {
  colours,
  scaffolding,
  transitionBuilder,
  materialBuilder
} from '../../theme';

interface IProps {
  maxHeight: string;
  selectColour: string;
}

interface EProps {
  backgroundColour?: string;
}

export const AccordionWrapper = styled.div({
  label: 'acorrdion-wrapper',
  position: 'relative',
  width: '100%',
  boxShadow: `${materialBuilder(1)}`,
  backgroundColor: colours.white
});

export const AccordionInput = styled.input<IProps>(
  {
    label: 'accordion-input',
    position: 'absolute',
    top: '-10px',
    left: '-10px',
    display: 'none',

    ':checked + label': {
      color: colours.greyMid,
      '&:after': {
        content: `""`
      }
    }
  },
  ({ maxHeight, selectColour }) => ({
    ':checked + label + div': {
      maxHeight: maxHeight
    },
    ':checked + label': {
      color: colours.greyMid,
      '&:after': {
        backgroundColor: selectColour
      }
    }
  })
);

export const AccordionLabel = styled.label({
  label: 'accordion-label',
  display: 'flex',
  alignItems: 'center',
  minHeight: '56px',
  padding: `0px ${scaffolding.gutterLg}`,
  cursor: 'pointer',
  boxSizing: 'border-box',
  borderBottom: `1px solid ${colours.greyLight}`,
  color: colours.greyUltraLight,
  transition: `${transitionBuilder('color')}`,
  WebkitTapHighlightColor: colours.transparent,
  '&:after': {
    content: `""`,
    position: 'absolute',
    right: scaffolding.gutterLg,
    width: '16px',
    height: '16px',
    borderRadius: '100%',
    backgroundColor: colours.greyUltraLight,
    transition: `${transitionBuilder('background-color')}`
  }
});

export const AccordionContent = styled.div<EProps>(
  {
    label: 'accordion-content',
    overflow: 'hidden',
    maxHeight: '0px',
    backgroundColor: colours.offWhite,
    transition: `${transitionBuilder('max-height')}`
  },
  ({ backgroundColour }) => ({
    backgroundColor: backgroundColour
  })
);

export const AccordionContentInner = styled.div({
  padding: `${scaffolding.gutterLg}`,
  color: colours.greyMid
});
