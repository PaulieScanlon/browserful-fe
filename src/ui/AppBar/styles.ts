import styled from 'react-emotion';

import { colours, common, scaffolding, materialBuilder } from '../../theme';
interface IProps {
  fixed?: boolean;
  width?: string | null;
}

export const AppBarWrapper = styled.header<IProps>(
  {
    label: 'app-bar-wrapper',
    display: 'flex',
    position: 'fixed',
    top: '0px',
    zIndex: common.appBar.zIndex,
    alignItems: 'center',
    justifyContent: 'space-between',
    backgroundColor: colours.white,
    boxShadow: `${materialBuilder(1)}`,
    minHeight: common.appBar.height,
    padding: `0px ${scaffolding.gutterLg}`
  },
  ({ fixed, width }) => ({
    position: fixed ? 'fixed' : 'relative',
    width: width
  })
);
