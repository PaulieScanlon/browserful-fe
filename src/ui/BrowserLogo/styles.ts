import styled from 'react-emotion';
import { scaffolding } from '../../theme';

export const BrowserImage = styled.img({
  label: 'browser-image',
  width: '38px',
  height: 'auto',
  margin: `${scaffolding.gutterSm} ${scaffolding.gutterSm} ${
    scaffolding.gutterSm
  } 0px`
});
