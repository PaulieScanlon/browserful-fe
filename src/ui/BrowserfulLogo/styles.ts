import styled from 'react-emotion';
import { colours, scaffolding } from '../../theme';
import { font } from '../Typography';

export const LogoWrapper = styled.div({
  label: 'logo-wrapper',
  display: 'inline-flex',
  position: 'relative',
  alignItems: 'center'
});

export const SvgWrapper = styled.span({
  label: 'svg-wrapper',
  display: 'inline-flex',
  marginRight: '5px'
});

export const BetaBadge = styled.span({
  label: 'beta-badge',
  borderRadius: '4px',
  padding: '2px 4px',
  marginTop: '2px',
  marginLeft: scaffolding.gutterSm,
  color: colours.white,
  fontFamily: font.fontFamily,
  fontSize: '10px',
  lineHeight: '12px',
  backgroundColor: colours.pink
});
