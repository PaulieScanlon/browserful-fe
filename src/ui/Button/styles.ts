import styled from 'react-emotion';

import { scaffolding, colours } from '../../theme';
import { font } from '../../ui/Typography';

interface IProps {
  fontColour: string;
  backgroundColour: string;
  grow: boolean;
  disabled: boolean;
}

export const ButtonElement = styled.button<IProps>(
  {
    label: 'button-element',
    display: 'flex',
    justifyContent: 'center',
    padding: `${scaffolding.gutterSm} ${scaffolding.gutterLg}`,
    fontSize: font.fontSize,
    fontFamily: font.fontFamily,
    border: 0,
    borderRadius: '5px',
    outline: 'none'
  },
  ({ fontColour, backgroundColour, grow, disabled }) => ({
    color: fontColour,
    backgroundColor: disabled ? colours.greyUltraLight : backgroundColour,
    width: grow ? '100%' : 'auto',
    cursor: disabled ? 'not-allowed' : 'pointer'
  })
);
