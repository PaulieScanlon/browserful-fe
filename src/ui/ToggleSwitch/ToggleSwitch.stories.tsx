import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';

import { ToggleSwitch } from './ToggleSwitch';
import { colours } from '../../theme';

const onChange = event => {
  action('onChange')(event.currentTarget.id);
};

const stories = storiesOf('ToggleSwitch', module);

stories.add(
  'default usage',
  withInfo(
    'ToggleSwitch renders an HTML label wrapping an input checkbox. An id is required'
  )(() => <ToggleSwitch id="storybook-togggle-switch" />)
);

stories.add(
  'onChange',
  withInfo('The onChange handler passes the synthetic event')(() => (
    <ToggleSwitch
      id="storybook-togggle-switch"
      onChange={event => onChange(event)}
    />
  ))
);

stories.add(
  'type',
  withInfo(
    'the type prop can be used so the ToggleSwitch can act as a radio input'
  )(() => <ToggleSwitch id="storybook-togggle-switch" type="radio" />)
);

stories.add(
  'defaultChecked',
  withInfo(
    'The defaultChecked is used to start an ToggleSwtich checked/selected'
  )(() => <ToggleSwitch id="storybook-togggle-switch" defaultChecked />)
);

stories.add(
  'selectColour',
  withInfo('The selectColour controls the colour of the checkbox background')(
    () => (
      <ToggleSwitch id="storybook-togggle-switch" selectColour={colours.teal} />
    )
  )
);

stories.add(
  'theme',
  withInfo(
    'The theme prop converts the colour theme from default light to dark'
  )(() => <ToggleSwitch id="storybook-togggle-switch" theme="dark" />)
);

stories.add(
  'children',
  withInfo('Children can be used to render label text')(() => (
    <ToggleSwitch id="storybook-togggle-switch">Label text</ToggleSwitch>
  ))
);

stories.add(
  'renderContent',
  withInfo(
    'renderContent can be used to render a sibling element with config styles for checked and unchecked'
  )(() => (
    <ToggleSwitch
      id="storybook-togggle-switch"
      renderContent={{
        config: {
          checked: {
            transition: '.2s linear',
            color: colours.pink
          },
          unchecked: {
            transition: '.2s linear',
            color: colours.greyLight
          }
        },
        component: <div>Toggle my styles</div>
      }}
    >
      Label text
    </ToggleSwitch>
  ))
);

stories.add(
  'flexDirection',
  withInfo('flexDirection is any valid css flex-direction value')(() => (
    <ToggleSwitch flexDirection="row-reverse" id="storybook-togggle-switch">
      Label text
    </ToggleSwitch>
  ))
);

stories.add(
  'justifyContent',
  withInfo('...')(() => (
    <ToggleSwitch justifyContent="space-between" id="storybook-togggle-switch">
      Label text
    </ToggleSwitch>
  ))
);

stories.add(
  'functional test',
  withInfo('Testing width with row-reverse')(() => (
    <div
      style={{
        display: 'flex',
        width: '200px',
        border: `1px solid ${colours.greyUltraLight}`
      }}
    >
      <ToggleSwitch
        justifyContent="space-between"
        flexDirection="row-reverse"
        id="storybook-togggle-switch"
        defaultChecked
      >
        Label text
      </ToggleSwitch>
    </div>
  ))
);
