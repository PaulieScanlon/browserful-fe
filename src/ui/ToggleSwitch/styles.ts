import styled from 'react-emotion';
import { transitionBuilder, colours, scaffolding } from '../../theme';
import { font } from '../../ui/Typography';
interface IProps {
  selectColour?: string;
  flexDirection?: string;
  justifyContent?: string;
  config?: {
    checked: any;
    unchecked: any;
  };
}

export const SwitchLabel = styled.label<IProps>(
  {
    label: 'switch-label',
    margin: `${scaffolding.gutterSm} 0px`,
    cursor: 'pointer',
    userSelect: 'none',
    WebkitTapHighlightColor: colours.transparent
  },
  ({ flexDirection }: any) => ({
    flexDirection: flexDirection
  })
);

export const SwitchInput = styled.input<IProps>(
  {
    label: 'switch-input',
    position: 'absolute',
    top: '-10px',
    left: '-10px',
    display: 'none',
    transition: transitionBuilder('background-color'),

    ':checked + .switch-controls': {
      '.switch-slider': {
        ':before': {
          transform: 'translateX(14px)'
        }
      }
    }
  },
  ({ selectColour, config }) => ({
    ':checked + .switch-controls': {
      '.switch-slider': {
        '+ .switch-text': {
          color: colours.greyMid
        },
        backgroundColor: selectColour
      }
    },
    ':checked ~ .switch-content': {
      ...config.checked
    },
    ' ~ .switch-content': {
      ...config.unchecked
    },
    ':hover + .switch-controls': {
      '.switch-slider': {
        backgroundColor: selectColour
      }
    }
  })
);

export const SwitchSlider = styled.span({
  label: 'switch-slider',
  position: 'relative',
  display: 'inline-block',
  borderRadius: '32px',
  width: '32px',
  height: '18px',
  right: '0px',
  backgroundColor: colours.greyUltraLight,
  transition: transitionBuilder('all'),
  ':before': {
    position: 'absolute',
    content: `""`,
    borderRadius: '50%',
    width: '12px',
    height: '12px',
    left: '3px',
    bottom: '3px',
    backgroundColor: colours.white,
    transition: transitionBuilder('all')
  }
});

export const SwitchText = styled.small({
  label: 'switch-text',
  fontSize: `${font.fontSize}`,
  lineHeight: `${font.lineHeight}`,
  fontFamily: `${font.fontFamily}`,
  fontWeight: 'normal',
  textAlign: 'center',
  margin: `0px ${scaffolding.gutterSm}`,
  color: colours.greyLight,
  transition: transitionBuilder('all')
});

export const SwitchControls = styled.div<IProps>(
  {
    label: 'switch-controls',
    display: 'flex',
    alignItems: 'center'
  },
  ({ justifyContent }) => ({
    justifyContent: justifyContent
  })
);

export const SwitchContent = styled.div({
  label: 'switch-content',
  display: 'flex'
});
