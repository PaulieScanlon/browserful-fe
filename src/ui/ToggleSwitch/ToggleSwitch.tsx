import * as React from 'react';

import { colours } from '../../theme';
import { SwitchLabel, SwitchInput, SwitchSlider, SwitchText } from './styles';

interface IProps {
  id: string;
  name?: string;
  type?: string;
  defaultChecked?: boolean;
  selectColour?: string;
  flexDirection?: string;
  children?: React.ReactChild;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

export const ToggleSwitch: React.SFC<IProps> = ({
  id,
  name,
  type,
  defaultChecked,
  selectColour,
  flexDirection,
  onChange,
  children
}: IProps) => {
  return (
    <SwitchLabel htmlFor={id} flexDirection={flexDirection}>
      <SwitchInput
        id={id}
        name={name}
        type={type}
        defaultChecked={defaultChecked}
        selectColour={selectColour}
        onChange={event => onChange(event)}
      />
      <SwitchSlider className="switch-slider" />
      <SwitchText className="switch-text">{children}</SwitchText>
    </SwitchLabel>
  );
};

ToggleSwitch.defaultProps = {
  defaultChecked: false,
  type: 'checkbox',
  selectColour: colours.pink,
  flexDirection: 'row',
  onChange: () => {}
};
