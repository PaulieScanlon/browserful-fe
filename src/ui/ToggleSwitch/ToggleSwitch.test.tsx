import * as React from 'react';
import { shallow, mount } from 'enzyme';

import { ToggleSwitch } from './ToggleSwitch';
import { SwitchInput } from './styles';

// const onChange = (event, browser, version) => {};

const shallowDefault = props => {
  return shallow(<ToggleSwitch id="test-togggle-switch" {...props} />);
};

describe('<ToggleSwitch />', () => {
  it('renders correctly', () => {
    const wrapper = shallowDefault({});
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders as checkbox', () => {
    const wrapper = shallowDefault({});
    expect(wrapper.find(SwitchInput).props().type).toEqual('checkbox');
  });
  it('fires onChange', () => {
    const testOnChange = jest.fn();
    const wrapper = mount(
      <ToggleSwitch
        id="test-togggle-switch"
        onChange={event => testOnChange(event)}
      />
    );

    const switchInput = wrapper.find(SwitchInput);
    switchInput.simulate('change');
    expect(testOnChange).toHaveBeenCalledTimes(1);
  });
});
