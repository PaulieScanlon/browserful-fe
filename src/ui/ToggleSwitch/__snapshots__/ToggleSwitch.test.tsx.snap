// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToggleSwitch
    defaultChecked={false}
    flexDirection="row"
    id="test-togggle-switch"
    justifyContent="flex-start"
    onChange={[Function]}
    selectColour="#ff5ca3"
    theme="light"
    type="checkbox"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <SwitchInput
          config={
            Object {
              "checked": Object {},
              "unchecked": Object {},
            }
          }
          defaultChecked={false}
          id="test-togggle-switch"
          onChange={[Function]}
          selectColour="#ff5ca3"
          type="checkbox"
        />,
        <SwitchLabel
          className="switch-label"
          htmlFor="test-togggle-switch"
        >
          <SwitchControls
            className="switch-controls"
            flexDirection="row"
            justifyContent="flex-start"
          >
            <SwitchSlider
              className="switch-slider"
              theme="light"
            />
            <span
              style={
                Object {
                  "width": "12px",
                }
              }
            />
            <SwitchText
              className="switch-text"
            />
          </SwitchControls>
        </SwitchLabel>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "config": Object {
            "checked": Object {},
            "unchecked": Object {},
          },
          "defaultChecked": false,
          "id": "test-togggle-switch",
          "name": undefined,
          "onChange": [Function],
          "selectColour": "#ff5ca3",
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <SwitchControls
              className="switch-controls"
              flexDirection="row"
              justifyContent="flex-start"
            >
              <SwitchSlider
                className="switch-slider"
                theme="light"
              />
              <span
                style={
                  Object {
                    "width": "12px",
                  }
                }
              />
              <SwitchText
                className="switch-text"
              />
            </SwitchControls>,
            undefined,
          ],
          "className": "switch-label",
          "htmlFor": "test-togggle-switch",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <SwitchSlider
                  className="switch-slider"
                  theme="light"
                />,
                <span
                  style={
                    Object {
                      "width": "12px",
                    }
                  }
                />,
                <SwitchText
                  className="switch-text"
                />,
              ],
              "className": "switch-controls",
              "flexDirection": "row",
              "justifyContent": "flex-start",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "switch-slider",
                  "theme": "light",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "style": Object {
                    "width": "12px",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": undefined,
                  "className": "switch-text",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          undefined,
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <SwitchInput
            config={
              Object {
                "checked": Object {},
                "unchecked": Object {},
              }
            }
            defaultChecked={false}
            id="test-togggle-switch"
            onChange={[Function]}
            selectColour="#ff5ca3"
            type="checkbox"
          />,
          <SwitchLabel
            className="switch-label"
            htmlFor="test-togggle-switch"
          >
            <SwitchControls
              className="switch-controls"
              flexDirection="row"
              justifyContent="flex-start"
            >
              <SwitchSlider
                className="switch-slider"
                theme="light"
              />
              <span
                style={
                  Object {
                    "width": "12px",
                  }
                }
              />
              <SwitchText
                className="switch-text"
              />
            </SwitchControls>
          </SwitchLabel>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "config": Object {
              "checked": Object {},
              "unchecked": Object {},
            },
            "defaultChecked": false,
            "id": "test-togggle-switch",
            "name": undefined,
            "onChange": [Function],
            "selectColour": "#ff5ca3",
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <SwitchControls
                className="switch-controls"
                flexDirection="row"
                justifyContent="flex-start"
              >
                <SwitchSlider
                  className="switch-slider"
                  theme="light"
                />
                <span
                  style={
                    Object {
                      "width": "12px",
                    }
                  }
                />
                <SwitchText
                  className="switch-text"
                />
              </SwitchControls>,
              undefined,
            ],
            "className": "switch-label",
            "htmlFor": "test-togggle-switch",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <SwitchSlider
                    className="switch-slider"
                    theme="light"
                  />,
                  <span
                    style={
                      Object {
                        "width": "12px",
                      }
                    }
                  />,
                  <SwitchText
                    className="switch-text"
                  />,
                ],
                "className": "switch-controls",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "switch-slider",
                    "theme": "light",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "style": Object {
                      "width": "12px",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": undefined,
                    "className": "switch-text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            undefined,
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
